runtime/int.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
+#include "balrt.h"
+
+
+// lang.int:MIN_VALUE is the longest, which is 17 chars
+#define MAX_HEX_INT_LEN 17
+
+GC TaggedPtr _Bint__toHexString(int64_t i) {
+    int size = smallStringSize(MAX_HEX_INT_LEN);
+    SmallStringPtr p = _bal_alloc(size);
+    int len;
+    if (i < 0) {
+        // simply -i worked, but doing this to avoid UB
+        len = sprintf(p->bytes, "-%llx", ((uint64_t) -(i + 1)) + 1);
```

Comment:
`-(uint64_t)i` is, I believe, sufficient to avoid UB

Passing a `uint64_t` to a `%llx` isn't ideal, because there's nothing to say that `long long` is exactly 64 bits. I just had a look at this and it turns out there's a PRIx64 macro for this (from inttypes.h).

(I need to fix print.c to use this also.)